applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py

```diff
@@ -129,11 +130,31 @@ def RunOptimizationLoop(self):
             KM.Logger.PrintInfo("ShapeOpt", "Time needed for current optimization step = ", timer.GetLapTime(), "s")
             KM.Logger.PrintInfo("ShapeOpt", "Time needed for total optimization so far = ", timer.GetTotalTime(), "s")
 
+            if self.__isAlgorithmConverged():
+                break
+
     # --------------------------------------------------------------------------
     def FinalizeOptimizationLoop(self):
         self.analyzer.FinalizeAfterOptimizationLoop()
         self.data_logger.FinalizeDataLogging()
 
+    # --------------------------------------------------------------------------
+    def __isAlgorithmConverged(self):
+
+        if self.opt_iteration > 1 :
+            # Check if maximum iterations were reached
+            if self.opt_iteration == self.algorithm_settings["max_iterations"].GetInt():
```

Comment:
if this is tested here, the loop above could be
```python
while not self.__isAlgorithmConverged():
```

Reply:
Forget it then we would need to increment the counter, i prefer the way you did it.